GraphConsoleAppV3/GraphRequests.cs

```diff
@@ -0,0 +1,1534 @@
+                    await user.UpdateAsync();
+                    Console.WriteLine("User {0} is successfully UNassigned an app (role).", user.DisplayName);
+
+                }
+            }
+
+            catch (Exception e)
+            {
+                Program.WriteError("\nError Assigning Direct Permission: {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+        }
+
+        #endregion
+
+        private static async Task PrintDevices(IActiveDirectoryClient client)
+        {
+            #region Get Devices
+
+            //*********************************************************************************************
+            // Get a list of Mobile Devices from tenant
+            //*********************************************************************************************
+            Console.WriteLine("\nGetting Devices");
+            IPagedCollection<IDevice> devices = null;
+            try
+            {
+                devices = await client.Devices.ExecuteAsync();
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine("/nError getting devices {0} {1}", e.Message,
+                    e.InnerException != null ? e.InnerException.Message : "");
+            }
+
+            if (devices != null)
+            {
+                do
+                {
+                    List<IDevice> devicesList = devices.CurrentPage.ToList();
+                    foreach (IDevice device in devicesList)
+                    {
+                        if (device.ObjectId != null)
+                        {
+                            Console.WriteLine("Device ID: {0}, Type: {1}", device.DeviceId, device.DeviceOSType);
+                            IPagedCollection<IDirectoryObject> registeredOwners = device.RegisteredOwners;
+                            if (registeredOwners != null)
+                            {
+                                do
+                                {
+                                    List<IDirectoryObject> registeredOwnersList = registeredOwners.CurrentPage.ToList();
+                                    foreach (IDirectoryObject owner in registeredOwnersList)
+                                    {
+                                        Console.WriteLine("Device Owner ID: " + owner.ObjectId);
+                                    }
+                                    registeredOwners = await registeredOwners.GetNextPageAsync();
+                                } while (registeredOwners != null);
+                            }
+                        }
+                    }
+                    devices = await devices.GetNextPageAsync();
+                } while (devices != null);
+            }
+
+            #endregion
+        }
+
+        private static async Task CreateOAuth2Permission(IActiveDirectoryClient client, ServicePrincipal newServicePrincpal)
```

Comment:
>CreateOAuth2Permission [](start = 34, length = 22)

This method is doing creation and deletion of permission as well as deletion of service principal.
1) dont combine code dealing with different entities.
2) Name functions correctly to reflect correctly what the method is doing.